### ISSUE TITLE
[Youtube] `ios` player client: HTTP Error 400: Bad Request - Request contains an invalid argument 

### ISSUE THREAD

**Issue by All-The-Foxes** [yt-dlp/yt-dlp#11640]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

USA

### Provide a description that is worded well enough to be understood

I truncated the output past the 'downloading video' bit because it's not relevant. The downloads themselves work, but the iOS API can no longer be used. I just picked a random video. This will happen with anything from YouTube.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
<snip>@<snip>-linux:~$ yt-dlp --cookies-from-browser firefox -ciwvU -S "res:1440" -o "~/MEGA/Stuff/Videos/kitty/%(title)s.%(ext)s" https://www.youtube.com/watch?v=ltV4bhWJpuE
[debug] Command-line config: ['--cookies-from-browser', 'firefox', '-ciwvU', '-S', 'res:1440', '-o', '~/MEGA/Stuff/Videos/kitty/%(title)s.%(ext)s', 'https://www.youtube.com/watch?v=ltV4bhWJpuE']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2024.11.24.232931 from yt-dlp/yt-dlp-nightly-builds [4b5eec0aa] (pip)
[debug] Python 3.12.3 (CPython x86_64 64bit) - Linux-6.8.0-49-generic-x86_64-with-glibc2.39 (OpenSSL 3.0.13 30 Jan 2024, glibc 2.39)
[debug] exe versions: ffmpeg 6.1.1 (setts), ffprobe 6.1.1
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.06.02, mutagen-1.46.0, requests-2.32.3, sqlite3-3.45.1, urllib3-2.2.1, websockets-13.1
[debug] Proxy map: {}
Extracting cookies from firefox
[debug] Extracting cookies from: "/home/<snip>/snap/firefox/common/.mozilla/firefox/<snip>.default/cookies.sqlite"
Extracted 1570 cookies from firefox
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1837 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-nightly-builds/releases/latest
Latest version: nightly@2024.11.24.232931 from yt-dlp/yt-dlp-nightly-builds
yt-dlp is up to date (nightly@2024.11.24.232931 from yt-dlp/yt-dlp-nightly-builds)
[youtube] Extracting URL: https://www.youtube.com/watch?v=ltV4bhWJpuE
[youtube] ltV4bhWJpuE: Downloading webpage
[debug] [youtube] Extracted SAPISID cookie
[youtube] ltV4bhWJpuE: Downloading ios player API JSON
WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument.
WARNING: [youtube] HTTP Error 400: Bad Request. Retrying (1/3)...
[youtube] ltV4bhWJpuE: Downloading ios player API JSON
WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument.
WARNING: [youtube] HTTP Error 400: Bad Request. Retrying (2/3)...
[youtube] ltV4bhWJpuE: Downloading ios player API JSON
WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument.
WARNING: [youtube] HTTP Error 400: Bad Request. Retrying (3/3)...
[youtube] ltV4bhWJpuE: Downloading ios player API JSON
WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument.
WARNING: [youtube] Unable to download API page: HTTP Error 400: Bad Request (caused by <HTTPError 400: Bad Request>)
[youtube] ltV4bhWJpuE: Downloading mweb player API JSON
[debug] Loading youtube-nsig.89dfc5b3 from cache
[debug] [youtube] Decrypted nsig -1J1qrP4I7_VD_ => 8_Q3QczUUN_WIg
[debug] Loading youtube-nsig.89dfc5b3 from cache
[debug] [youtube] Decrypted nsig oTSCUTHuLdq2As => QL0QEIAVBw4rlA
[debug] Sort order given by user: res:1440
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, res:1440(1440.0), quality, fps, hdr:12(7), source, vcodec, channels, acodec, lang, proto, size, br, asr, vext, aext, hasaud, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] ltV4bhWJpuE: Downloading 1 format(s): 399+251
```

**Comment by bashonly** [comment]
Does it happen if you don't pass cookies?

**Comment by All-The-Foxes** [comment]
> Does it happen if you don't pass cookies?

Just tried that and no it doesn't, however it tells me to sign in to confirm my age despite the video not being age restricted.

**Comment by bashonly** [comment]
I haven't yet been able to reproduce this issue with or without passing cookies.

All that's known for certain is that this is an A/B test or slow rollout of some sort that breaks yt-dlp's usage of the `ios` player_client.

It *seems* like cookies are the culprit: it's possible that YT is disabling authentication via cookie with `ios` player requests.

But it's possible that this is a symptom an even larger problem: the change that's being A/B tested breaks yt-dlp's `ios` requests completely (regardless of whether or not cookies were passed), and this issue is only being noticed by users when passing cookies (and for now is only reproducible when passing cookies, since once it's rolled out for an account it stays rolled out).

Hopefully it's just the former.

**Comment by coletdjnz** [comment]
Also reported here https://github.com/coletdjnz/yt-dlp-youtube-oauth2/issues/48#issuecomment-2496343648

Given changes to the other ios endpoints where they blocked cookies, this is likely cookie related.

**Comment by xpalidotious** [comment]
I don't normally use the ios client while downloading but noticed today that when I occasionally tried to view the available formats for a video using the ios client that I consistently got this error,

I came here to to see if this had been reported and when I found nothing I just assumed it must be a problem on my end since most people will be using the ios client by default. Can also confirm that if I don't pass cookies then I can use the ios client successfully.

**Comment by VeggieJay** [comment]
Also have the same warnings, passing from cookies.txt extracted with the Chrome extension

**Comment by Rosemyne** [comment]
I'm having this issue as well. I've tried it twice either using the cookies-from-browser option or using the cookies-from-browser option to write the cookies to a cookies.txt file and passing that through manually. Both cases seem to result in a HTTP Error 400: Bad Requests.

Command 1:
yt-dlp.exe --write-subs --sub-langs en --sub-format best --remux-video mkv --embed-subs -f bestvideo*+bestaudio/best
--embed-thumbnail -o thumbnail:png --cookies cookies.txt URL

Command 2:
yt-dlp.exe --write-subs --sub-langs en --sub-format best --remux-video mkv --embed-subs -f bestvideo*+bestaudio/best --embed-thumbnail -o thumbnail:png --cookies-from-browser firefox:[profile]::[container] URL

``` 
yt-dlp.exe -vU --write-subs --sub-langs en --sub-format best --remux-video mkv --embed-subs -f bestvideo*+bestaudio/b
est --embed-thumbnail -o thumbnail:png --cookies cookies.txt https://www.youtube.com/watch?v=mHnt8TVbC9M
[debug] Command-line config: ['-vU', '--write-subs', '--sub-langs', 'en', '--sub-format', 'best', '--remux-video', 'mkv', '--embed-subs', '-f', 'bestvideo*+bestaudio/best', '--embed-thumbnail', '-o', 'thumbnail:png', '--cookies', 'cookies.txt', 'https://www.youtube.com/watch?v=mHnt8TVbC9M']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.03.10 from yt-dlp/yt-dlp [615a84447] (pip)
[debug] Python 3.8.2 (CPython AMD64 32bit) - Windows-10-10.0.19041-SP0 (OpenSSL 1.1.1d  10 Sep 2019)
[debug] exe versions: ffmpeg 4.3.1-2020-11-08-full_build-www.gyan.dev, ffprobe 4.3.1-2020-11-08-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2019.09.11, mutagen-1.45.1, requests-2.31.0, sqlite3-3.28.0, urllib3-2.2.1, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1803 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
[debug] Downloading _update_spec from https://github.com/yt-dlp/yt-dlp/releases/latest/download/_update_spec
Current version: stable@2024.03.10 from yt-dlp/yt-dlp
Latest version: stable@2024.11.18 from yt-dlp/yt-dlp
Upgradable to: stable@2024.10.22 from yt-dlp/yt-dlp
ERROR: You installed yt-dlp with pip or using the wheel from PyPi; Use that to update
[youtube] Extracting URL: https://www.youtube.com/watch?v=mHnt8TVbC9M
[youtube] mHnt8TVbC9M: Downloading webpage
[debug] [youtube] Extracted SAPISID cookie
[youtube] mHnt8TVbC9M: Downloading ios player API JSON
WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument.
WARNING: [youtube] HTTP Error 400: Bad Request. Retrying (1/3)...
[youtube] mHnt8TVbC9M: Downloading ios player API JSON
WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument.
WARNING: [youtube] HTTP Error 400: Bad Request. Retrying (2/3)...
[youtube] mHnt8TVbC9M: Downloading ios player API JSON
WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument.
WARNING: [youtube] HTTP Error 400: Bad Request. Retrying (3/3)...
[youtube] mHnt8TVbC9M: Downloading ios player API JSON
WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument.
WARNING: [youtube] Unable to download API page: HTTP Error 400: Bad Request (caused by <HTTPError 400: Bad Request>)
[youtube] mHnt8TVbC9M: Downloading android player API JSON
WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument.
WARNING: [youtube] HTTP Error 400: Bad Request. Retrying (1/3)...
[youtube] mHnt8TVbC9M: Downloading android player API JSON
WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument.
WARNING: [youtube] HTTP Error 400: Bad Request. Retrying (2/3)...
[youtube] mHnt8TVbC9M: Downloading android player API JSON
WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument.
WARNING: [youtube] HTTP Error 400: Bad Request. Retrying (3/3)...
[youtube] mHnt8TVbC9M: Downloading android player API JSON
WARNING: [youtube] YouTube said: ERROR - Request contains an invalid argument.
WARNING: [youtube] Unable to download API page: HTTP Error 400: Bad Request (caused by <HTTPError 400: Bad Request>)
[debug] [youtube] Extracting signature function js_89dfc5b3_106
[youtube] mHnt8TVbC9M: Downloading player 89dfc5b3
[debug] Saving youtube-sigfuncs.js_89dfc5b3_106 to cache
WARNING: [youtube] mHnt8TVbC9M: nsig extraction failed: You may experience throttling for some formats
         n = abDKUt2okrXuBa ; player = https://www.youtube.com/s/player/89dfc5b3/player_ias.vflset/en_US/base.js
[debug] [youtube] Unable to extract nsig function code (caused by RegexNotFoundError('Unable to extract \x1b[0;94mInitial JS player n function name\x1b[0m; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U')); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
WARNING: [youtube] mHnt8TVbC9M: nsig extraction failed: You may experience throttling for some formats
         n = M_fRQuY2Fm4qTb ; player = https://www.youtube.com/s/player/89dfc5b3/player_ias.vflset/en_US/base.js
[debug] [youtube] Extracting signature function js_89dfc5b3_102
[debug] Saving youtube-sigfuncs.js_89dfc5b3_102 to cache
[info] mHnt8TVbC9M: Downloading subtitles: en
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, size, br, asr, vext, aext, hasaud, id
[info] mHnt8TVbC9M: Downloading 1 format(s): 248+251
[info] Writing video subtitles to: ツユ - アンダーヒロイン MV [mHnt8TVbC9M].en.vtt
[debug] Invoking http downloader on "https://www.youtube.com/api/timedtext?v=mHnt8TVbC9M&ei=3NlFZ62lNbWX2_gPvI_P2AU&caps=asr&opi=112496729&exp=xbt&xoaf=5&hl=en&ip=0.0.0.0&ipbits=0&expire=1732656204&sparams=ip%2Cipbits%2Cexpire%2Cv%2Cei%2Ccaps%2Copi%2Cexp%2Cxoaf&signature=39127A93608BCF70AAE601ED7B534964B3A32A86.CAFF0E0CDD275B4D53A969C28BD3615EE1F0B3C5&key=yt8&lang=en&fmt=vtt"
[debug] File locking is not supported. Proceeding without locking
[download] Destination: ツユ - アンダーヒロイン MV [mHnt8TVbC9M].en.vtt
[download] 100% of    4.81KiB in 00:00:00 at 44.16KiB/s
[info] Downloading video thumbnail 41 ...
[info] Writing video thumbnail 41 to: ツユ - アンダーヒロイン MV [mHnt8TVbC9M].webp
WARNING: webm doesn't support embedding a thumbnail, mkv will be used
[debug] Invoking http downloader on "https://rr2---sn-vgqsrnzy.googlevideo.com/videoplayback?expire=1732652604&ei=3NlFZ62lNbWX2_gPvI_P2AU&ip=12.36.67.19&id=o-AGoWoEq8RXDHqQcomg0Bnq57BBaGfIva3xKzi-tg3JGV&itag=248&aitags=133%2C134%2C135%2C136%2C137%2C160%2C242%2C243%2C244%2C247%2C248%2C278%2C394%2C395%2C396%2C397%2C398%2C399&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&met=1732631004%2C&mh=QX&mm=31%2C26&mn=sn-vgqsrnzy%2Csn-p5qs7n6d&ms=au%2Conr&mv=m&mvi=2&pl=18&rms=au%2Cau&initcwndbps=851250&siu=1&bui=AQn3pFQ7aCDBhvYeyM-FNHkOiavX7hCMsZjElWHtwIWIpm2Zq3ewBAu_v3Xn6A_g2jP6_fEvFw&spc=qtApAVeWCBxtquuszDxxXik9q-fLm7q9Zfrjx-kt0SagprT053mJcmc4i80jkgUakH0cjMYllA&vprv=1&svpuc=1&mime=video%2Fwebm&ns=X-4PvHgyzs89g1BNG55aYHQQ&rqh=1&gir=yes&clen=39927470&dur=168.600&lmt=1726266074130621&mt=1732630414&fvip=2&keepalive=yes&fexp=51326932%2C51335594%2C51355912&c=WEB&sefc=1&txp=4535434&n=M_fRQuY2Fm4qTb&sparams=expire%2Cei%2Cip%2Cid%2Caitags%2Csource%2Crequiressl%2Cxpc%2Csiu%2Cbui%2Cspc%2Cvprv%2Csvpuc%2Cmime%2Cns%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&lsparams=met%2Cmh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Crms%2Cinitcwndbps&lsig=AGluJ3MwRAIgJgGhW3q64TWgW-H4_LIo0Z7-SV9qI0L6ixUXUR8HIZoCIHecC4Yz7W_T3ufhi5LFmumn-jtg7Mp5rrwGM4tXDRFP&sig=AJfQdSswRQIhAIUgG8Er_wh_BVeVV6-TjSYHoFSYnM4bFvPooBOFheg7AiBNnMp0trcYlTyDoHjBXF29aibTAlHrZ1rU3aVyqrjYug=="
ERROR: unable to download video data: HTTP Error 403: Forbidden
Traceback (most recent call last):
  File "C:\Users\UserAccount\AppData\Local\Programs\Python\Python38-32\lib\site-packages\yt_dlp\YoutubeDL.py", line 3416, in process_info
    partial_success, real_download = self.dl(fname, new_info)
  File "C:\Users\UserAccount\AppData\Local\Programs\Python\Python38-32\lib\site-packages\yt_dlp\YoutubeDL.py", line 3155, in dl
    return fd.download(name, new_info, subtitle)
  File "C:\Users\UserAccount\AppData\Local\Programs\Python\Python38-32\lib\site-packages\yt_dlp\downloader\common.py", line 455, in download
    ret = self.real_download(filename, info_dict)
  File "C:\Users\UserAccount\AppData\Local\Programs\Python\Python38-32\lib\site-packages\yt_dlp\downloader\http.py", line 369, in real_download
    establish_connection()
  File "C:\Users\UserAccount\AppData\Local\Programs\Python\Python38-32\lib\site-packages\yt_dlp\downloader\http.py", line 120, in establish_connection
    ctx.data = self.ydl.urlopen(request)
  File "C:\Users\UserAccount\AppData\Local\Programs\Python\Python38-32\lib\site-packages\yt_dlp\YoutubeDL.py", line 4101, in urlopen
    return self._request_director.send(req)
  File "C:\Users\UserAccount\AppData\Local\Programs\Python\Python38-32\lib\site-packages\yt_dlp\networking\common.py", line 115, in send
    response = handler.send(request)
  File "C:\Users\UserAccount\AppData\Local\Programs\Python\Python38-32\lib\site-packages\yt_dlp\networking\_helper.py", line 204, in wrapper
    return func(self, *args, **kwargs)
  File "C:\Users\UserAccount\AppData\Local\Programs\Python\Python38-32\lib\site-packages\yt_dlp\networking\common.py", line 326, in send
    return self._send(request)
  File "C:\Users\UserAccount\AppData\Local\Programs\Python\Python38-32\lib\site-packages\yt_dlp\networking\_requests.py", line 351, in _send
    raise HTTPError(res, redirect_loop=max_redirects_exceeded)
yt_dlp.networking.exceptions.HTTPError: HTTP Error 403: Forbidden
```

**Comment by bashonly** [comment]
@Rosemyne you are using an outdated version of yt-dlp. Update and you should still be able to get mweb formats.

**Comment by bashonly** [comment]
The TL;DR of this issue is:

### Using cookies with the `ios` player_client is broken

- If you need `ios` formats (e.g. format 616), then **don't pass cookies**

- **Pass cookies only when necessary** (e.g. age-restricted videos, members-only videos)

- If you need to pass cookies, then you can **still get `mweb` formats**. If your yt-dlp is up-to-date, it will extract them by default. If you want to avoid the ios warnings, pass `--extractor-args "youtube:player_client=default,-ios"` with your command.

- If you are passing cookies to yt-dlp ***and*** passing `--extractor-args "youtube:player_client=ios"` (or an equivalent arg value like `default,-mweb`), then you'll need to stop passing one of them
